subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/DefaultFileSystemLocationFingerprint.java

```diff
@@ -45,8 +50,73 @@ private static HashCode hashForType(FileType fileType, HashCode hash) {
         }
     }
 
+    @Override
+    public final void appendToHasher(Hasher hasher) {
+        hasher.putString(getNormalizedPath());
+        hasher.putHash(getNormalizedContentHash());
+    }
+
+    @Override
+    public FileType getType() {
+        return type;
+    }
+
     @Override
     public String getNormalizedPath() {
         return normalizedPath;
     }
+
+    @Override
+    public HashCode getNormalizedContentHash() {
+        return normalizedContentHash;
+    }
+
+    @Override
+    public final int compareTo(FileSystemLocationFingerprint o) {
+        int result = getNormalizedPath().compareTo(o.getNormalizedPath());
+        if (result == 0) {
+            result = getNormalizedContentHash().compareTo(o.getNormalizedContentHash());
+        }
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Consider simplifying these using `Objects`:
```java
@Override
public boolean equals(Object o) {
    if (this == o) {
        return true;
    } else if (o == null || getClass() != o.getClass()) {
        return false;
    }

    DefaultFileSystemLocationFingerprint that = (DefaultFileSystemLocationFingerprint) o;
    return Objects.equals(type, that.type)
        && Objects.equals(normalizedContentHash, that.normalizedContentHash)
        && Objects.equals(normalizedPath, that.normalizedPath);
}

@Override
public int hashCode() {
    return Objects.hash(type, normalizedContentHash, normalizedPath);
}
```

Reply:
We usually use IntelliJ's built-in `equals()`/`hashCode()` generation that works well. Given that we have that pattern I don't see enough benefit here to deviate from that pattern for this particular case.